pkg/binder/binder_unit.go

```diff
@@ -267,79 +267,50 @@ func (unitInfo *bindingUnitInfo) GetNewTasksOnNode(nodeName string) []*runningUn
 	return result
 }
 
-func (unitInfo *bindingUnitInfo) GetReadyAndWaitingTasks() []*checkResult {
-	unitInfo.mu.Lock()
-	defer unitInfo.mu.Unlock()
-
-	len := len(unitInfo.readyTasks) + len(unitInfo.waitingTasks)
-	if len == 0 {
+func (unitInfo *bindingUnitInfo) getTasks(tasks map[types.UID]*checkResult) []*checkResult {
+	length := len(tasks)
+	if length == 0 {
 		return nil
 	}
 
-	result := make([]*checkResult, len)
+	result := make([]*checkResult, length)
 	i := 0
-	for _, cr := range unitInfo.readyTasks {
-		result[i] = cr
-		i++
-	}
-	for _, cr := range unitInfo.waitingTasks {
+	for _, cr := range tasks {
 		result[i] = cr
 		i++
 	}
 	return result
 }
 
-func (unitInfo *bindingUnitInfo) GetReadyTasks() []*checkResult {
+func (unitInfo *bindingUnitInfo) GetReadyAndWaitingTasks() []*checkResult {
 	unitInfo.mu.Lock()
 	defer unitInfo.mu.Unlock()
-
-	len := len(unitInfo.readyTasks)
-	if len == 0 {
-		return nil
+	allTasks := make(map[types.UID]*checkResult)
```

Comment:
no need to allocate `allTasks`, just return as follows:

```
unitInfo.mu.Lock()
defer unitInfo.mu.Unlock()

return append(unitInfo.getTasks(unitInfo.readyTasks), unitInfo.getTasks(unitInfo.waitingTasks)...)
```

Reply:
yeah, it look good to me, thanks!

Reply:
done